src/compute-client/src/controller/instance.rs

```diff
@@ -899,7 +937,14 @@ where
             .filter(|(id, _)| self.compute.collections.contains_key(id))
             .collect();
 
-        self.update_write_frontiers(replica_id, &updates).await
+        self.update_write_frontiers(replica_id, &updates).await?;
+
+        let dropped_collections: Vec<_> = updates
+            .iter()
+            .filter(|(_, frontier)| frontier.is_empty())
+            .map(|(id, _)| id.clone())
+            .collect();
+        self.update_dropped_collections(dropped_collections)
```

Comment:
If `update_write_frontiers` already checks for dropped collections, why do we need to do it here again?

Reply:
I think that `update_write_frontiers` only checks for dropped collections conditionally on read capability changes. But what I am trying to cover for here is a change in the `replica_write_frontiers` of the collection.